File: app/src/main/kotlin/com/ufkoku/ciphersharedpreferences/demo/DemoApp.java
package com.ufkoku.ciphersharedpreferences.demo;

import android.app.Application;

import com.ufkoku.ciphersharedpreferences.demo.dagger2.app_context.AppContextModule;

public class DemoApp extends Application {

    private static DemoApp instance;

    private AppContextModule appContextModule;

    @Override
    public void onCreate() {
        appContextModule = new AppContextModule(this);

        instance = this;

        super.onCreate();
    }

    public static DemoApp getInstance() {
        return instance;
    }

    public static AppContextModule getAppContextModule() {
        return instance.appContextModule;
    }

}

File: app/src/main/kotlin/com/ufkoku/ciphersharedpreferences/demo/DemoApp.kt
package com.ufkoku.ciphersharedpreferences.demo

import android.app.Application

import com.ufkoku.ciphersharedpreferences.demo.dagger2.app_context.AppContextModule

class DemoApp : Application() {

    companion object {

        var instance: DemoApp? = null
            private set

        fun getAppContextModule(): AppContextModule? {
            return instance!!.appContextModule
        }

    }

    private var appContextModule: AppContextModule? = null

    override fun onCreate() {
        appContextModule = AppContextModule(this)

        instance = this

        super.onCreate()
    }

}
